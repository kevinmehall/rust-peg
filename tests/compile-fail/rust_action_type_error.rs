extern crate peg;

struct X;
struct Y;

peg::parser!(grammar foo() for &str {
    rule foo() -> X = "a" { Y } //~ ERROR
});

fn main() {}
