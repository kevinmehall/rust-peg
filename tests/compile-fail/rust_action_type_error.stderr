error[E0308]: mismatched types
 --> tests/compile-fail/rust_action_type_error.rs:7:27
  |
7 |     rule foo() -> X = "a" { Y } //~ ERROR
  |                           ^^^^^
  |                           |
  |                           expected struct `X`, found struct `Y`
  |                           arguments to this enum variant are incorrect
  |
help: the type constructed contains `Y` due to the type of the argument passed
  --> tests/compile-fail/rust_action_type_error.rs:7:27
   |
7  |     rule foo() -> X = "a" { Y } //~ ERROR
   |                           ^^^^^ this argument influences the type of `{{root}}`
note: tuple variant defined here
  --> peg-runtime/lib.rs
   |
   |     Matched(usize, T),
   |     ^^^^^^^
   = note: this error originates in the macro `peg::parser` (in Nightly builds, run with -Z macro-backtrace for more info)
